File: src/app/src/environment.ts
function getEnvVar<T>(key: string, defaultValue?: T): T {
  let value = process.env[key] as T | undefined;

  if (typeof value === "undefined") {
    value = import.meta.env[key] as T | undefined;
  }

  if (typeof value === "undefined" && typeof defaultValue !== "undefined") {
    value = defaultValue;
  }

  if (typeof value === "undefined") {
    throw new Error(`${key} is not defined.`);
  }

  return value as T;
}

export function getEnv() {
  return {
    openRouterApiKey: getEnvVar<string>("OPENROUTER_API_KEY"),
    openRouterTitle: getEnvVar<string>("OPENROUTER_TITLE", "Logdog"),
    openRouterReferer: getEnvVar<string>("OPENROUTER_REFERER", "https://dennise.me"),
    openRouterModel: getEnvVar<string>("OPENROUTER_MODEL", "moonshotai/kimi-k2.5"),
  };
}

export const ENVIRONMENT = getEnvVar<"production" | "development">("NODE_ENV", "development");
export const isProduction = ENVIRONMENT === "production";
export const isDevelopment = ENVIRONMENT === "development";

File: src/server/src/environment.py
import os
from dotenv import load_dotenv, find_dotenv
from pydantic import SecretStr

load_dotenv(find_dotenv())


def _get_env_var(key: str, defaultValue: str | None = None) -> SecretStr:
    value = os.environ.get(key)
    if value is None:
        value = defaultValue
    if value is None:
        raise ValueError(f"Environment variable '{key}' is not defined.")
    value = value.strip().strip("'\"").strip()
    return SecretStr(value)


SECRET_KEY = _get_env_var("SECRET_KEY")

DATABASE_URL = _get_env_var("DATABASE_URL")
MEGABASE_URL = _get_env_var("MEGABASE_URL")

OPENROUTER_API_KEY = _get_env_var("OPENROUTER_API_KEY")
OPENROUTER_TITLE = _get_env_var("OPENROUTER_TITLE", "Logdog")
OPENROUTER_REFERER = _get_env_var("OPENROUTER_REFERER", "https://dennise.me")
OPENROUTER_MODEL = _get_env_var("OPENROUTER_MODEL", "openrouter/auto")

STORAGE_URL = _get_env_var("STORAGE_URL", "http://localhost:3000/assets")
